source/reactor/inc/reactor/EventHandlers.h

```diff
@@ -0,0 +1,68 @@
+#ifndef Reactor_EventHandlers_H
+#define Reactor_EventHandlers_H
+
+#include <functional>
+
+namespace u {
+namespace reactor {
+class EventLoop;
+
+/**
+ * @brief Event handler function definition.
+ *
+ * Parameters to the event handler function are:
+ * - The file descriptor of the OS resource where the event was originated at. Event handler can
+ * perform, for example, accept, read, write and close operations on the file descriptor.
+ * - The event loop where it was caught. Event handler can install new handlers, remove existing
+ * handlers and even ask for loop termination.
+ */
+using EventHandler = std::function<void(int, EventLoop *)>;
+
+/**
+ * Set of event handler functions, one function to handle each possible event happened.
+ * Handlers are grouped into a struct to facilitate passing them as a single parameter to the install and
+ * remove handlers functions of the event loop.
```

Comment:
Yes. I prefer having a context for the function when it is called, some state it can query, some other structure it can grab information from. Not expecting everything to be passed with the parameters.
This can be achieved for example if `on_read()`, `on_write()`, etc. are part of an object that can be initialized with the required data.